src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -175,7 +180,7 @@ private IEnumerable<TDocument> DeserializeDocuments(IEnumerable<RawBsonDocument>
 
             foreach (var rawDocument in rawDocuments)
             {
-                if (!rawDocument.Contains("_id"))
+                if (!rawDocument.Contains("_id") && _maxWireVersion < 8)
```

Comment:
This change is not necessary.

If maxWireVersion >= 8 the server would have returned an error and we would never reach this line of code.